pkg/sirenia/state/state.go

```diff
@@ -147,15 +147,20 @@ func (x *PgConfig) IsNewDownstream(y *PgConfig) bool {
 	return y.Downstream != nil && !peersEqual(x.Downstream, y.Downstream)
 }
 
-type Postgres interface {
+type Database interface {
+	// TODO(jpg): need a more generic representation of database log position.
+	// the only important thing here is to ensure we have an ordering.
+	// the comparison of this should also be handed of to the underlying database wrapper
 	XLogPosition() (xlog.Position, error)
-	Reconfigure(*PgConfig) error
+	// Returns the implementation of the XLog interface for this database
+	XLog() xlog.XLog
```

Comment:
@josephglanville At one point we discussed using an interface that supported comparison returned from `XLogPosition` instead of the singleton, as the only reason it is singleton is `Zero` which is only used be the simulator. Not sure if this is still the case, but wanted to point it out.